### ISSUE TITLE
Sound.play() SounChannel.state  stackoverflow

### ISSUE THREAD

**Issue by ahmedhosnypro** [korlibs/korge#2251]
the problem appear  when trying to get the state of SoundChanell that results from Sound.play()

in the following  each state, playing, onComplete calls causes stackoverflow
```kotlin
            val c= sound.play()
            c.state 
            c.playing
            c.onCompleted(scope.coroutineContext) {
                //..
            }
```

but with the following there's no problem at all
```kotlin
            val sound = nativeSoundProvider.createSound(data = bytes)
            val audioStream = sound.toStream()
            val soundAudioStream = SoundAudioStream(
                stream = audioStream,
                coroutineContext = scope.coroutineContext,
                soundProvider = nativeSoundProvider
            )

            val channel = soundAudioStream.platformPlay()

            playerSoundChannel = channel
            channel.onCompleted(scope.coroutineContext) {
                playerSoundChannel = null
            }
```

korge version = "6.0.0-alpha5"
tested on: Jvm , android

**Comment by soywiz** [comment]
Fixed in `6.0.0-beta1`

**Comment by ahmedhosnypro** [comment]
> Fixed in `6.0.0-beta1`

thank you
